From 26b6dc53796121eb01dc7212b218844426e54ca0 Mon Sep 17 00:00:00 2001
From: Jorge Ruesga <jorge@ruesga.com>
Date: Sun, 29 Mar 2015 03:44:57 +0200
Subject: [PATCH 02/19] email: fix back button

If conversation_topmost_overlay is not ready to take the focus, if the drawerlayout
has the focus (this happens always that the drawerlayout is shown in a conversation view)
then the onBackPressed event never get dispatched to the mail activity, and press
the back button doesn't have effect.
Just double check that the conversation_topmost_overlay view can take focus. In case it
can't take the focus don't response we handled the keyup event.

Also, add a double check over the drawer slide event to prevent burger menu to be displayed
when the back arrow button should be the one displayed.

Change-Id: I964eb1eb779af13c9b2c07b77049147c2ff1f2d9
Signed-off-by: Jorge Ruesga <jorge@ruesga.com>
---
 src/com/android/mail/ui/ConversationViewFragment.java | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/com/android/mail/ui/ConversationViewFragment.java b/src/com/android/mail/ui/ConversationViewFragment.java
index 0f52cc705..0ccfb8233 100644
--- a/src/com/android/mail/ui/ConversationViewFragment.java
+++ b/src/com/android/mail/ui/ConversationViewFragment.java
@@ -1255,10 +1255,7 @@ public class ConversationViewFragment extends AbstractConversationViewFragment i
 
             // Finally we handle the special keys
             if (keyCode == KeyEvent.KEYCODE_BACK && id != R.id.conversation_topmost_overlay) {
-                if (isActionUp) {
-                    mTopmostOverlay.requestFocus();
-                }
-                return true;
+                return mTopmostOverlay.requestFocus();
             } else if (keyCode == KeyEvent.KEYCODE_ENTER &&
                     id == R.id.conversation_topmost_overlay) {
                 if (isActionUp) {
-- 
2.17.1

